### PR TITLE
fix(checkbox): upgrade mdc-web to v1

### PR DESCRIPTION
**Pull request by gugu**

### CODE REVIEW

**Comment by moog16** [comment]
related #697

**Comment by gugu** [comment]
image

It was already implemented before

File: packages/checkbox/index.tsx [hunk 1]
```diff
@@ -45,20 +45,22 @@ interface CheckboxState {
...
 
   constructor(props: CheckboxProps) {
     super(props);
     this.state = {
       'checked': props.checked,
       'indeterminate': props.indeterminate,
       'classList': new Set(),
-      'aria-checked': false,
+      'aria-checked': 'false',
+      'disabled': props.disabled!,
```
Comment (moog16) [inline packages/checkbox/index.tsx]: what is this new state.disabled variable for?

**Comment by moog16** [comment]
Sorry I didn't realize that we had already update checkbox :)

File: packages/checkbox/index.tsx (hunk 1 above)
Comment (gugu) [reply packages/checkbox/index.tsx]: Adapter now has `setNativeControlDisabled: (disabled) => this.updateState('disabled', disabled),`, as I understand mdc-web now can change disabled state

Comment (moog16) [reply packages/checkbox/index.tsx]: ah ok - but we don't use it in the render method.

Comment (gugu) [reply packages/checkbox/index.tsx]: done

File: packages/checkbox/index.tsx [hunk 2]
```diff
@@ -91,7 +94,10 @@ export class Checkbox extends React.Component<CheckboxProps, CheckboxState> {
       this.handleChange(checked!, indeterminate!);
     }
     if (disabled !== prevProps.disabled) {
-      this.foundation.setDisabled(disabled);
+      this.setState({
```
Comment (moog16) [inline packages/checkbox/index.tsx]: @gugu I dont think we need to setState here. foundation.setDisabled() will call `setNativeControlDisabled`, which updates disabled for us.

File: packages/checkbox/package.json [hunk 3]
```diff
@@ -16,7 +16,7 @@
     "url": "https://github.com/material-components/material-components-web-react.git"
   },
   "dependencies": {
-    "@material/checkbox": "^0.41.0",
+    "@material/checkbox": "^1.0.0",
```
Comment (moog16) [inline packages/checkbox/package.json]: 1.1.0

**Pull request merged**